zuul-core/src/main/java/com/netflix/zuul/message/http/HttpRequestMessageImpl.java

```diff
@@ -528,6 +532,15 @@ public int getOriginalPort()
         return port;
     }
 
+    @Override
+    public int getClientDestinationPort() {
```

Comment:
This may be absent, so the API should express this.   It can either be an `Optional<Integer>` or a `@Nullable Integer`

Reply:
I chose to fallback to the implementation for `getOriginalPort`. In that case, the port should not have those values.
We could modify the API to `Optional` if that makes more sense, as opposed to a best effort port value.